### ISSUE TITLE
Bedrock Init Error

### ISSUE THREAD

**Issue by raviprrasadcg** [smartcontracts/simple-optimism-node#75]
Receiving the following error while starting the Docker containers:

bedrock-init_1            | ++ echo ''
bedrock-init_1            | + BEDROCK_FLAG=
bedrock-init_1            | + '[' '' == true ']'
bedrock-init_1            | + echo 'Bedrock flag not set, waiting...'
bedrock-init_1            | + sleep 60
bedrock-init_1            | Bedrock flag not set, waiting...

**Comment by zainirfan13** [comment]
Receiving same error and l2geth container is stuck after "verifier loop"
I dont understand why would you upgrade if you couldn't provide a good guide to upgrade.

++ config bedrock/mainnet/ready
+++ cast call 0xcbebc5ba53ff12165239cbb3d310fda2236d6ad2 'config(address,string)(string)' 0x68108902De3A5031197a6eB3b74b3b033e8E8e4d bedrock/mainnet/ready --rpc-url https://goerli.infura.io/v3/84842078b09946638c03157f83405213
++ echo ''
+ BEDROCK_FLAG=
+ '[' '' == true ']'
+ echo 'Bedrock flag not set, waiting...'
+ sleep 60
Bedrock flag not set, waiting...
+ true

**Comment by germashevanton** [comment]
same issue

**Comment by raviprrasadcg** [comment]
@zainirfan13 @germashevanton You may want to use the following docker image until the issue gets resolved:

`https://github.com/CryptoManufaktur-io/optimism-docker/tree/main`

I have used the above docker config and the node is up now.

**Comment by Salexber** [comment]
> @zainirfan13 @germashevanton You may want to use the following docker image until the issue gets resolved:
> 
> `https://github.com/CryptoManufaktur-io/optimism-docker/tree/main`
> 
> I have used the above docker config and the node is up now.

How to setup .env for bedrock mainnet?

**Comment by raviprrasadcg** [comment]
Just rename the `default.env` within the above repo to `.env`. Then update the following parameters:

`NETWORK=mainnet`
`SNAPSHOT=https://storage.googleapis.com/oplabs-mainnet-data/mainnet-bedrock.tar`
`L1_RPC=PLACE_YOUR_ALCHEMY_URL_HERE`

You can leave the other options to its default value. It takes around 2 hours (depending on your Internet connection speed) to complete the node bootstrap.

**Comment by zhwindy** [comment]
> Receiving same error and l2geth container is stuck after "verifier loop" I dont understand why would you upgrade if you couldn't provide a good guide to upgrade.
> 
> ++ config bedrock/mainnet/ready +++ cast call 0xcbebc5ba53ff12165239cbb3d310fda2236d6ad2 'config(address,string)(string)' 0x68108902De3A5031197a6eB3b74b3b033e8E8e4d bedrock/mainnet/ready --rpc-url https://goerli.infura.io/v3/84842078b09946638c03157f83405213 ++ echo ''
> 
> * BEDROCK_FLAG=
> * '[' '' == true ']'
> * echo 'Bedrock flag not set, waiting...'
> * sleep 60
>   Bedrock flag not set, waiting...
> * true

the same issue too, It feels like this simple project is getting more and more complicated

**Comment by zainirfan13** [comment]
> raviprrasadcg

Will this serve as a optimism node or I will have to run the other containers of simple-optimism-node ?  with it ?

Thanks a lot

**Comment by zainirfan13** [comment]
INFO [06-09|14:06:38.733] Generating state snapshot                root=bfe2b0..c2d21c in=d8fa7c..bcf59f at=c33c46..e17f8a accounts=108,313 slots=7,429,828 storage=593.81MiB dangling=0 elapsed=2m40.002s   eta=28.775s
INFO [06-09|14:06:46.733] Generating state snapshot                root=bfe2b0..c2d21c in=f41d26..e9ffb7 at=6e6bc4..4b19e5 accounts=121,756 slots=7,814,194 storage=626.95MiB dangling=0 elapsed=2m48.002s   eta=8.18s
INFO [06-09|14:06:50.271] Generated state snapshot                 accounts=127,608 slots=7,992,368 storage=641.73MiB dangling=0 elapsed=2m51.539s


These are the logs i see when i ran the docker of https://github.com/CryptoManufaktur-io/optimism-docker/tree/main @raviprrasadcg

**Comment by raviprrasadcg** [comment]
@zainirfan13 The above output is expected. It takes lot of time to complete the process. I was in touch with the Optimism user group in the Discord and they suggested me to turn off the snapshot process temporarily to speed up the process:

`It can happen, I recommend you use --snapshot=false as op-geth flag until it has synced to the latest part of the chain, to not get interrupted with snapshot work while catching up`

So, I have added the following in the env:

`EL_EXTRAS=--snapshot=false`

and restarted the containers.

**Comment by zainirfan13** [comment]
after following you instruction
now all I see in the logs of optimism-docker-op-geth-1 is 

```
INFO [06-09|14:52:53.459] Loaded most recent local block           number=4,061,224 hash=0f7835..05175f td=0 age=4mo3w6d
INFO [06-09|14:52:53.459] Loaded most recent local finalized block number=4,061,224 hash=0f7835..05175f td=0 age=4mo3w6d
INFO [06-09|14:52:53.459] Initialising Ethereum protocol           network=420 dbversion=8
INFO [06-09|14:52:53.461] Loaded local transaction journal         transactions=0 dropped=0
INFO [06-09|14:52:53.461] Regenerated local transaction journal    transactions=0 accounts=0
INFO [06-09|14:52:53.462] Chain post-merge, sync via beacon client 
INFO [06-09|14:52:53.462] Gasprice oracle is ignoring threshold set threshold=2
WARN [06-09|14:52:53.462] Engine API enabled                       protocol=eth
INFO [06-09|14:52:53.462] Starting peer-to-peer node               instance=Geth/v0.1.0-unstable-0a77db9c/linux-amd64/go1.20.5
INFO [06-09|14:52:53.520] IPC endpoint opened                      url=/var/lib/op-geth/geth.ipc
INFO [06-09|14:52:53.520] Loaded JWT secret file                   path=/var/lib/op-geth/ee-secret/jwtsecret crc32=0xa15c614b
INFO [06-09|14:52:53.520] New local node record                    seq=5 id=d02ca800c6957ef8 ip=127.0.0.1 udp=0 tcp=30303
INFO [06-09|14:52:53.520] Started P2P networking                   self="enode://966efad9d4f21debf82f3e33e29ca938318ecd75941bc76264a8c89490575cd6d38240a6141695c900f5bdccd2f7ab1d9ddcdf407b171d983f7ac894eca5dba8@127.0.0.1:30303?discport=0"
INFO [06-09|14:52:53.521] HTTP server started                      endpoint=[::]:8545 auth=false prefix= cors=* vhosts=*
INFO [06-09|14:52:53.521] WebSocket enabled                        url=ws://[::]:8546
INFO [06-09|14:52:53.521] WebSocket enabled                        url=ws://[::]:8551
INFO [06-09|14:52:53.521] HTTP server started                      endpoint=[::]:8551 auth=true  prefix= cors=localhost vhosts=*
```

**Comment by raviprrasadcg** [comment]
You need to wait for sometime. It'll start downloading the chain segments.

**Comment by zainirfan13** [comment]
For those who are still struggling to run the mainnet node, I managed to run the node by pulling the latest code on branch 
sc/bedrock 

I followed the README.md and did the required changes. One thing more I did was add "restart: unless-stopped" in the docker-compose.yml file for "bedrock-init:" because it kept on loosing connection when download bedrock.tar snapshot which was about 400 GB. Luckily it resumed everytime the docker of bedrock init restarted, I was able to download it in 3 to 4 hours. 

The JSONRPC port is 9993 and ws port is 9994 by default.

**Comment by luoxiandong** [comment]
> @zainirfan13 The above output is expected. It takes lot of time to complete the process. I was in touch with the Optimism user group in the Discord and they suggested me to turn off the snapshot process temporarily to speed up the process:
> 
> `It can happen, I recommend you use --snapshot=false as op-geth flag until it has synced to the latest part of the chain, to not get interrupted with snapshot work while catching up`
> 
> So, I have added the following in the env:
> 
> `EL_EXTRAS=--snapshot=false`
> 
> and restarted the containers.

<img width="1300" alt="image" src="https://github.com/smartcontracts/simple-optimism-node/assets/19297244/0d75c1a3-f376-445c-a47d-6b152cc22c74">
I do it,but node has not been synced for 12 hours.
`optimism-docker-op-geth-1  | INFO [06-13|16:48:35.499] HTTP server started                      endpoint=[::]:8545 auth=false prefix= cors=* vhosts=*
optimism-docker-op-geth-1  | INFO [06-13|16:48:35.499] WebSocket enabled                        url=ws://[::]:8546
optimism-docker-op-geth-1  | INFO [06-13|16:48:35.499] WebSocket enabled                        url=ws://[::]:8551
optimism-docker-op-geth-1  | INFO [06-13|16:48:35.499] HTTP server started                      endpoint=[::]:8551 auth=true  prefix= cors=localhost vhosts=*
optimism-docker-op-geth-1  | INFO [06-13|17:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-13|17:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-13|17:48:35.493] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed=13.242ms
optimism-docker-op-geth-1  | INFO [06-13|18:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-13|18:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-13|18:48:35.480] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed="951.081µs"
optimism-docker-op-geth-1  | INFO [06-13|19:48:35.480] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-13|19:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-13|19:48:35.481] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed="933.794µs"
optimism-docker-op-geth-1  | INFO [06-13|20:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-13|20:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-13|20:48:35.480] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed=1.160ms
optimism-docker-op-geth-1  | INFO [06-13|21:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-13|21:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-13|21:48:35.480] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed="962.808µs"
optimism-docker-op-geth-1  | INFO [06-13|22:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-13|22:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-13|22:48:35.480] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed="981.97µs"
optimism-docker-op-geth-1  | INFO [06-13|23:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-13|23:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-13|23:48:35.481] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed=1.065ms
optimism-docker-op-geth-1  | INFO [06-14|00:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-14|00:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-14|00:48:35.481] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed=1.072ms
optimism-docker-op-geth-1  | INFO [06-14|01:48:35.479] Writing clean trie cache to disk         path=/var/lib/op-geth/geth/triecache      threads=1
optimism-docker-op-geth-1  | INFO [06-14|01:48:35.480] Regenerated local transaction journal    transactions=0 accounts=0
optimism-docker-op-geth-1  | INFO [06-14|01:48:35.481] Persisted the clean trie cache           path=/var/lib/op-geth/geth/triecache      elapsed=1.110ms`
@raviprrasadcg

**Comment by raviprrasadcg** [comment]
@luoxiandong The above process took me around 2 days and 15 hours to complete.

**Comment by luoxiandong** [comment]
> @luoxiandong The above process took me around 2 days and 15 hours to complete.

tks!

**Comment by bmalepaty** [comment]
> https://github.com/CryptoManufaktur-io/optimism-docker/tree/main

Can you please help with detailed steps on how you used the CryptoNamufactur repo to get the node up? Thanks in advance

**Comment by Chomtana** [comment]
Hi, my fork has fixed this issue and uses an official snapshot.

Check out https://github.com/Chomtana/simple-optimism-node/tree/main/scripts

Pull request https://github.com/smartcontracts/simple-optimism-node/pull/97

**Comment by smartcontracts** [comment]
Closed by #97
